src/openloco/windows/CompanyWindow.cpp

```diff
@@ -0,0 +1,1501 @@
+            make_widget({ 3, 44 }, { 96, 120 }, widget_type::viewport, 1, -2),
+            make_widget({ 0, 0 }, { 24, 24 }, widget_type::wt_9, 1, string_ids::null, string_ids::move_main_view_to_show_this),
+            make_widget({ 178, 57 }, { 66, 66 }, widget_type::wt_9, 1, string_ids::null),
+            make_widget({ 154, 124 }, { 112, 22 }, widget_type::wt_9, 1, string_ids::null, string_ids::tooltip_change_owner_name),
+            widget_end(),
+        };
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << widx::centre_on_viewport) | (1 << widx::face) | (1 << widx::change_owner_name);
+
+        static window_event_list events;
+
+        // 0x00431EBB
+        static void prepare_draw(window* self)
+        {
+            printf("In prepare draw for company window %p with number = %d\n", self, self->number);
+
+            common::switchTabWidgets(self);
+
+            // Set company name.
+            auto company = companymgr::get(self->number);
+            *_common_format_args = company->name;
+
+            self->disabled_widgets &= ~((1 << widx::centre_on_viewport) | (1 << widx::face));
+
+            // No centering on a viewport that doesn't exist.
+            // if (self->viewports[0] == nullptr)
+            //     self->disabled_widgets |= (1 << widx::centre_on_viewport);
+
+            // No changing other player's faces, unless we're editing a scenario.
+            if (self->number != companymgr::get_controlling_id() && !is_editor_mode())
+                self->disabled_widgets |= (1 << widx::face);
+
+            self->widgets[common::widx::frame].right = self->width - 1;
+            self->widgets[common::widx::frame].bottom = self->height - 1;
+
+            self->widgets[common::widx::panel].right = self->width - 1;
+            self->widgets[common::widx::panel].bottom = self->height - 1;
+
+            self->widgets[common::widx::caption].right = self->width - 2;
+
+            self->widgets[common::widx::close_button].left = self->width - 15;
+            self->widgets[common::widx::close_button].right = self->width - 3;
+
+            self->widgets[widx::viewport].right = self->width - 119;
+            self->widgets[widx::viewport].bottom = self->height - 14;
+
+            self->widgets[widx::unk_11].top = self->height - 12;
+            self->widgets[widx::unk_11].bottom = self->height - 3;
+            self->widgets[widx::unk_11].right = self->width - 14;
+
+            self->widgets[widx::change_owner_name].right = self->width - 4;
+            self->widgets[widx::change_owner_name].left = self->width - 116;
+
+            self->widgets[widx::face].right = self->width - 28;
+            self->widgets[widx::face].left = self->width - 93;
+
+            self->widgets[common::widx::company_select].right = self->width - 3;
+            self->widgets[common::widx::company_select].left = self->width - 28;
+
+            if (self->number != companymgr::get_controlling_id())
```

Comment:
Would be more clear if switched around. `if (owner) { enable } else { disable }`.

Also, I'm not sure about if/else statements without curly braces.